react-backend/client/api_client.js

```diff
@@ -0,0 +1,241 @@
+/*
+ *
+ */
+import axios from 'axios';
+import { zipWith } from 'ramda';
+
+const parseAgeFilter = (ageBrackets) => {
+  const agesLower = (ageBrackets[0] !== 0 ? [0] : []).concat(ageBrackets);
+  const agesUpper = agesLower
+    .map(x => x - 1)
+    .filter(x => x > 0)
+    .concat(Infinity);
```

Comment:
This is an unnecessary use of `Infinity`, you can improve readability of this by using:
```
const parseAgeFilter = (ageBrackets) => {
  const agesLower = (ageBrackets[0] !== 0 ? [0] : []).concat(ageBrackets);

  const agesUpper = agesLower
    .map(x => x - 1)
    .filter(x => x > 0)
    .concat("false");

  return zipWith((x, y) => (y!=="false" ? `${x}-${y}` : `${x}-more`), agesLower, agesUpper)[0];
};
```

Reply:
Disagree that this is more readable. Leaving aside the use of a string value that reads like a boolean, `"false"` is not an age, and so is out of place in `agesUpper`.

I may have agreed had you suggested something like:
```js
const agesUpper = agesLower
  .map(x => x - 1)
  .filter(x => x > 0)
  .concat('more');

return zipWith((x, y) => `${x}-${y}`, agesLower, agesUpper)
```

Because it allows us to remove the branch from the final line. Although note that this is mixing presentational concerns with data transformations; the `'more'` is how we represent an undefined upper bound using text.

Maybe you would prefer:
```js
const ageToStr = n => Number.isFinite(n) ? `${n}` : 'more'

const agesLower = (ageBrackets[0] !== 0 ? [0] : [])
  .concat(ageBrackets)
  .map(ageToStr);

const agesUpper = agesLower
  .map(x => x - 1)
  .filter(x => x > 0)
  .concat(Infinity)
  .map(ageToStr);

return zipWith((x, y) => (`${x}-${y}`), agesLower, agesUpper)[0];
```

Reply:
> Disagree that this is more readable. Leaving aside the use of a string value that reads like a boolean, "false" is not an age, and so is out of place in agesUpper.

But following this logic, `Infinity` is also not an age.

How about:

```erlang
const parseAgeFilter = (ageBrackets) => {
  const agesLower = (ageBrackets[0] !== 0 ? [0] : []).concat(ageBrackets);
  
  const agesUpper = agesLower
    .map(x => x - 1)
    .filter(x => x > 0)
    .concat("okImSorryIllTryToStopMakingJokingInPRsMaybeINeedASignOrSomethingButITHoughtAStringifiedFalseWasEnough");

  return zipWith((x, y) => (typeof y !== 'String' ? `${x}-${y}` : `${x}-more`), agesLower, agesUpper)[0];
};
```

Reply:
But on a serious note, what's the expected params/output for this function? I feel like I can get similar functionality from:
```js
const parseAgeFilter = (ageBrackets) => 
  ageBrackets[0]<= 1 ? `0 - ${ageBrackets[1]-1}` : `0 - ${ageBrackets[0]-1}`
```

Reply:
I will eventually remove the `[0]` at the end of the return statement when we make changes to the API. Right now, the endpoint expects only one age-group (I believe), but we can make it more general.

Reply:
> But following this logic, Infinity is also not an age.

Let's say an "age" is any finite positive integer (a natural number). What we are concerned with is describing a set of intervals over the whole natural number line. All but the last interval is trivial. So we are looking a semantic to represent the final interval which might loosely be described as "anything greater than N, inclusive". The appropriate semantic for this in mathematics is a half-open interval `[N, Inf)`. So `Infinity`, while not strictly speaking an age, is the appropriate value to use here.

Reply:
**Disclaimer:** I'm not suggesting any of the code should be changed _but..._

... with the majority of time in dev world being spent on maintaining code (thus making readability financially beneficial), and the fair assumption that the majority of web developers do not have maths degrees, I'm merely pointing out that this is an "unusual" use of `Infinity` through poorly placed humour.